File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    queue = []

    # 각 작업이 완료되는데 걸리는 일수를 계산하여 큐에 저장
    for i in range(len(progresses)):
        progress = progresses[i]
        speed = speeds[i]
        days = (100 - progress) // speed
        if (100 - progress) % speed != 0:
            days += 1
        queue.append(days)

    # 첫 번째 기능부터 배포되는 기능 수 계산
    cnt = 1
    front = queue.pop(0)
    while queue:
        if front >= queue[0]:
            cnt += 1
            queue.pop(0)
        else:
            answer.append(cnt)
            cnt = 1
            front = queue.pop(0)
    answer.append(cnt)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    const n = speeds.length
    const days = new Array(n).fill()
    for(let i=0; i<n; i++){
        const leftOver = Math.ceil((100-progresses[i])/speeds[i])
        days[i] = leftOver
    }
    console.log(days)
    let cnt = 1 
    let current = days.shift()
    while (days.length > 0){
        if (current >= days[0]){
            cnt+=1
            days.shift()
        }
        else{
            answer.push(cnt)
            cnt = 1
            current = days.shift()
        }
    }
    answer.push(cnt)
    console.log(answer)
    return answer;
}